diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee25365..40a1936 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -171,6 +171,7 @@ endif()
 # Example binaries
 #============================================================================
 
+if (0)
 add_executable(example test/example.c)
 target_link_libraries(example zlib)
 add_test(example example)
@@ -188,3 +189,4 @@ if(HAVE_OFF64_T)
     target_link_libraries(minigzip64 zlib)
     set_target_properties(minigzip64 PROPERTIES COMPILE_FLAGS "-D_FILE_OFFSET_BITS=64")
 endif()
+endif()
