diff --git a/scripts/toolchains/linux.cmake b/scripts/toolchains/linux.cmake
index f22c71076..a68d6cbe8 100644
--- a/scripts/toolchains/linux.cmake
+++ b/scripts/toolchains/linux.cmake
@@ -13,7 +13,7 @@ elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
    string(APPEND VCPKG_LINKER_FLAGS " -m32")
 elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
     set(CMAKE_SYSTEM_PROCESSOR armv7l CACHE STRING "")
-    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64")
         if(NOT DEFINED CMAKE_CXX_COMPILER)
             set(CMAKE_CXX_COMPILER "arm-linux-gnueabihf-g++")
         endif()
@@ -30,7 +30,7 @@ elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
     endif()
 elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
     set(CMAKE_SYSTEM_PROCESSOR aarch64 CACHE STRING "")
-    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64")
         if(NOT DEFINED CMAKE_CXX_COMPILER)
             set(CMAKE_CXX_COMPILER "aarch64-linux-gnu-g++")
         endif()
